src/js/popper.js

```diff
@@ -0,0 +1,1210 @@
+      r = n ? E(t) : a(t, i(o));
+    return b(o, r, n);
+  }
+  function S(e) {
+    var t = e.ownerDocument.defaultView,
+      o = t.getComputedStyle(e),
+      n = parseFloat(o.marginTop || 0) + parseFloat(o.marginBottom || 0),
+      i = parseFloat(o.marginLeft || 0) + parseFloat(o.marginRight || 0),
+      r = { width: e.offsetWidth + i, height: e.offsetHeight + n };
+    return r;
+  }
+  function T(e) {
+    var t = { left: "right", right: "left", bottom: "top", top: "bottom" };
+    return e.replace(/left|right|bottom|top/g, function (e) {
+      return t[e];
+    });
+  }
+  function C(e, t, o) {
+    o = o.split("-")[0];
+    var n = S(e),
+      i = { width: n.width, height: n.height },
+      r = -1 !== ["right", "left"].indexOf(o),
+      p = r ? "top" : "left",
+      s = r ? "left" : "top",
+      d = r ? "height" : "width",
+      a = r ? "width" : "height";
+    return (
+      (i[p] = t[p] + t[d] / 2 - n[d] / 2),
+      (i[s] = o === s ? t[s] - n[a] : t[T(s)]),
+      i
+    );
+  }
+  function D(e, t) {
+    return Array.prototype.find ? e.find(t) : e.filter(t)[0];
+  }
+  function N(e, t, o) {
+    if (Array.prototype.findIndex)
+      return e.findIndex(function (e) {
+        return e[t] === o;
+      });
+    var n = D(e, function (e) {
+      return e[t] === o;
+    });
+    return e.indexOf(n);
+  }
+  function P(t, o, n) {
+    var i = void 0 === n ? t : t.slice(0, N(t, "name", n));
+    return (
+      i.forEach(function (t) {
+        t["function"] &&
+          console.warn("`modifier.function` is deprecated, use `modifier.fn`!");
+        var n = t["function"] || t.fn;
+        t.enabled &&
+          e(n) &&
+          ((o.offsets.popper = g(o.offsets.popper)),
+          (o.offsets.reference = g(o.offsets.reference)),
+          (o = n(o, t)));
+      }),
+      o
+    );
+  }
+  function k() {
+    if (!this.state.isDestroyed) {
+      var e = {
+        instance: this,
+        styles: {},
+        arrowStyles: {},
+        attributes: {},
+        flipped: !1,
+        offsets: {},
+      };
+      (e.offsets.reference = L(
+        this.state,
+        this.popper,
+        this.reference,
+        this.options.positionFixed
+      )),
+        (e.placement = O(
+          this.options.placement,
+          e.offsets.reference,
+          this.popper,
+          this.reference,
+          this.options.modifiers.flip.boundariesElement,
+          this.options.modifiers.flip.padding
+        )),
+        (e.originalPlacement = e.placement),
+        (e.positionFixed = this.options.positionFixed),
+        (e.offsets.popper = C(this.popper, e.offsets.reference, e.placement)),
+        (e.offsets.popper.position = this.options.positionFixed
+          ? "fixed"
+          : "absolute"),
+        (e = P(this.modifiers, e)),
+        this.state.isCreated
+          ? this.options.onUpdate(e)
+          : ((this.state.isCreated = !0), this.options.onCreate(e));
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/hello_100/jca-ui-kit/pullRequest?prid=5731813)